src/main/java/com/powsybl/caseserver/server/ObjectStorageService.java

```diff
@@ -0,0 +1,436 @@
+    }
+
+    @Override
+    public Optional<Network> loadNetwork(UUID caseUuid) {
+
+        if (!caseExists(caseUuid)) {
+            return Optional.empty();
+        }
+
+        return Optional.of(withS3DownloadedTempPath(caseUuid, path -> {
+            Network network = Network.read(path);
+            if (network == null) {
+                throw CaseException.createFileNotImportable(path);
+            }
+            return network;
+        }));
+    }
+
+    @Override
+    public void deleteCase(UUID caseUuid) {
+        String prefixKey = uuidToPrefixKey(caseUuid);
+        List<ObjectIdentifier> objectsToDelete = s3Client.listObjectsV2(builder -> builder.bucket(bucketName).prefix(prefixKey))
+            .contents()
+            .stream()
+            .map(s3Object -> ObjectIdentifier.builder().key(s3Object.key()).build())
+            .toList();
+
+        if (!objectsToDelete.isEmpty()) {
+            DeleteObjectsRequest deleteObjectsRequest = DeleteObjectsRequest.builder()
+                .bucket(bucketName)
+                .delete(Delete.builder().objects(objectsToDelete).build())
+                .build();
+            s3Client.deleteObjects(deleteObjectsRequest);
+            caseInfosService.deleteCaseInfosByUuid(caseUuid.toString());
+            caseMetadataRepository.deleteById(caseUuid);
+        }
+    }
+
+    @Override
+    public void deleteAllCases() {
+        ListObjectsV2Request listObjectsRequest = ListObjectsV2Request.builder()
+                .bucket(bucketName)
+                .prefix("blalala/")
+                .build();
+
+        ListObjectsV2Response listObjectsResponse = s3Client.listObjectsV2(listObjectsRequest);
+        List<ObjectIdentifier> objectsToDelete = listObjectsResponse.contents().stream()
+                .map(s3Object -> ObjectIdentifier.builder().key(s3Object.key()).build())
+                .toList();
+
+        if (!objectsToDelete.isEmpty()) {
+            DeleteObjectsRequest deleteObjectsRequest = DeleteObjectsRequest.builder()
+                    .bucket(bucketName)
+                    .delete(Delete.builder().objects(objectsToDelete).build())
+                    .build();
+
+            s3Client.deleteObjects(deleteObjectsRequest);
+        }
+
+        caseInfosService.deleteAllCaseInfos();
+        caseMetadataRepository.deleteAll();
+
+    }
+
+    @Override
+    public void setComputationManager(ComputationManager computationManager) {
+        this.computationManager = Objects.requireNonNull(computationManager);
+    }
+
+    /*
+     The query is an elasticsearch (Lucene) form query, so here it will be :
+     date:XXX AND geographicalCode:(X)
+     date:XXX AND geographicalCode:(X OR Y OR Z)
+    */
+
+    @Override
+    public List<CaseInfos> searchCases(String query) {
+        return caseInfosService.searchCaseInfos(query);
+    }
+
+    private void sendImportMessage(Message<String> message) {
+        OUTPUT_MESSAGE_LOGGER.debug("Sending message : {}", message);
```

Comment:
should not have rabbitmq code in the s3 file

Reply:
done